idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,361 @@
+property block(s) and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}"""))
+    CONFIG.declare("reaction_package", ConfigValue(
+        default=None,
+        # Removed domain argument due to limitations in ReactionBase for
+        # heterogeneous systems
+        description="Reaction package to use for unit",
+        doc="""Reaction parameter object used to define reaction calculations,
+**default** - None.
+**Valid values:** {
+**None** - no reaction package,
+**ReactionParameterBlock** - a ReactionParameterBlock object.}"""))
+    CONFIG.declare("reaction_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing reaction packages",
+        doc="""A ConfigBlock with arguments to be passed to a reaction block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see reaction package for documentation.}"""))
+
+    def build(self):
+        # Call UnitModel.build to setup dynamics
+        super(FixedBed0DData, self).build()
+
+        # Build Gas Phase StateBlock
+        # This block only needed so gas conc. to rxn block is calculated
+        # Defined state is set to True so that the "sum(mole_frac)=1" eqn in
+        # the gas state block is deactivated.
+        self.gas = self.config.gas_property_package.state_block_class(
+                self.flowsheet().time,
+                default={"parameters": self.config.gas_property_package,
+                         "defined_state": True})
+
+        # Build Solid Phase StateBlock
+        # As there is no solid flow, only need a single state block
+        # Defined state is set to True so that the "sum(mass_frac)=1" eqn in
+        # the solid state block is deactivated. This is done here as there is
+        # currently no way to deactivate the constraint at the initial time
+        # for batch systems (i.e. no inlet or outlet ports).
+        # The "sum(mass_frac)=1 for all t neq 0" eqn is written in the
+        # unit model instead
+        self.solids = self.config.solid_property_package.state_block_class(
+                self.flowsheet().time,
+                default={"parameters": self.config.solid_property_package,
+                         "defined_state": True})
+
+        tmp_dict = dict(**self.config.reaction_package_args)
+        tmp_dict["gas_state_block"] = self.gas
+        tmp_dict["solid_state_block"] = self.solids
+        tmp_dict["parameters"] = (self.config.reaction_package)
+        self.reactions = (
+                self.config.reaction_package.
+                reaction_block_class(
+                    self.flowsheet().time,
+                    doc="Reaction properties in control volume",
+                    default=tmp_dict))
+
+        # Volume of reactor
+        self.bed_diameter = Var(initialize=1,
+                                doc='Reactor diameter [m]',
+                                units=pyunits.m)
```

Comment:
Units for all these variables should be derived from the units defined in the property package, otherwise there are going to be unit consistency issues when someone tries to use a property package with different units.

There should also be unit conversion operations wherever two different property packages interact as there is no guarantee that both property packages use the same units).

Reply:
@andrewlee94 I defined variable units from the appropriate property package, in this case `units_meta_solid` which contains information from the solid properties. However, I'm running into a unit consistency error when running the tests - it seems that internally an added gas property is pressure in bar, while other properties use SI units (kJ, kg, m, s):

`E pyomo.core.base.units_container.InconsistentUnitsError: Error in units found in expression: s.gas_props.gas_const*fs.unit.gas 0.0].dens_mol*fs.unit.gas[0.0].temperature*0.01 - fs.unit.gas 0.0].pressure: kilojoule / meter ** 3 not compatible with bar.`

How would you recommend adding the unit conversion option to the FB0D model? The only interaction between gas/solid properties is the reaction package, and this error seems to be solely a gas-phase issue.

Reply:
I see a few things that could be wrong there.

1) The gas_constant might need to be converted from SI to be consistent with everything else (depends on how the gas constant is defined in the properties).
2. What is the `0.01` term, and does it have units as well?

You need to use `pyunits.convert(var_or_expr, to_units=units_of_properties)` wherever there is a possibility of units being different.

Reply:
The issues seem to be with the CLC property packages and not the unit model files. All of the regular IDAES tests pass with the updates to the FB0D initialization and unit definitions, and all four CLC flowsheet examples run and solve. However, all four flowsheets throw InconsistentUnitsError messages when queried with `assert_units_consistent(m)`. Since the FB0D model is working and all other tests pass, I propose finishing the FB0D files without the unit consistency test for now (MB and BFB don't have them either) and opening a new PR to add units to the property packages, and update the scaling and variable units in the MB and BFB models.

Some more detail on the specific errors I encountered:

When I check unit consistency for the examples, the reaction rate expression has a unit mismatch from a non-integer reaction order exponent. When I add an assert units test in the FB0D test file, I run into errors in the gas and solid properties. The one from above in the gas properties uses a dimensionless scaling factor instead of a `pyunit.convert` statement (kJ to J, then Pa to bar). After fixing the gas properties issue, the solid properties enthalpy Shomate expression runs into issues since the coefficients have no unit containers. I'll go through the gas, solid and reaction property packages and update the units to clear the issues.

Reply:
@andrewlee94 I'm onboard with @bpaul4 suggestion of completing this PR without the unit consistency test and subsequently opening a separate PR to add units to the property packages and update the scaling and variable units in the gas solid contactor models.

Reply:
Seeing as it sounds like there is some work required at the properties level first, that is probably the best idea. I just want to push this issue as I believe the gas-solids contactors are the only models that aren't using units and scaling at this point.